includes/specials/SpecialDeletedContributions.php

```diff
@@ -286,13 +301,19 @@ function execute( $par ) {
 		$options['limit'] = $request->getInt( 'limit', $wgQueryPageDefaultLimit );
 		$options['target'] = $target;
 
+		if ( User::idFromName( $target ) === null ) {
+			// User doesn't exist, so stop here
+			$out->addHTML( $this->getForm( $options ) );
+			$out->addWikiMsg( 'nocontribs' );
+			return;
+		}
+
 		$userObj = User::newFromName( $target, false );
 		if ( !$userObj ) {
 			$out->addHTML( $this->getForm( '' ) );
 			return;
 		}
 
-		$target = $userObj->getName();
 		$out->addSubtitle( $this->getSubTitle( $userObj ) );
 
 		if ( ( $ns = $request->getVal( 'namespace', null ) ) !== null && $ns !== '' ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "namespace" 10 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)